apis/betterself/v1/supplements/serializers.py

```diff
@@ -56,20 +56,32 @@ class IngredientCompositionReadOnlySerializer(serializers.Serializer):
 
 class IngredientCompositionCreateSerializer(serializers.Serializer):
     ingredient_uuid = serializers.UUIDField(source='ingredient.uuid')
-    measurement_uuid = serializers.UUIDField(source='measurement.uuid')
-    quantity = serializers.FloatField()
-
-    def create(self, validated_data):
-        user = self.context['request'].user
-        create_model = self.context['view'].model
+    measurement_uuid = serializers.UUIDField(source='measurement.uuid', required=False)
+    quantity = serializers.FloatField(required=False)
+    uuid = serializers.UUIDField(required=False, read_only=True)
 
+    def validate(self, validated_data):
```

Comment:
You realize you forgot that validate could be used as a step to do all the formatting and cleaning up prior to the create. Which is a good pattern to do ... so double-check all other serializers are doing this ??